.github/workflows/nightly.yaml

```
@@ -0,0 +1,39 @@
+name: Nightly
+on:
+  schedule:
+    # Times in UTC (PST+8), this translates to running Mon-Fri at midnight/1am (depending on DST)
+    - cron: 0 8 * * 1-5
+
+jobs:
+  plugins_test:
+    name: Nightly Plugin Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        linter-version: [Snapshots, Latest]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+        # TODO(Tyler): Determine if we want to wipe the download cache and start fresh each night.
+
+      - name: Cache tool downloads
+        uses: actions/cache@v3
+        with:
+          path: /tmp/plugins_testing_download_cache
+          # No need to key on trunk version unless we change how we store downloads.
+          key: trunk-${{ runner.os }}
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run plugin tests
+        run: |
+          PATH=$(npm bin):$PATH PLUGINS_TEST_LINTER_VERSION=${{ matrix.linter-version }} npm test --ci
```

i think using the native mechanism where possible is cleaner here:

```suggestion
      - name: Run plugin tests
        run: |
          PATH=$(npm bin):$PATH npm test --ci
        env:
          PLUGINS_TEST_LINTER_VERSION: ${{ matrix.linter-version }}
```